src/bwdist.jl

```diff
@@ -26,20 +30,45 @@ See also: [`distance_transform`](@ref).
 Transforms of Binary Images in Arbitrary Dimensions' [Maurer et al.,
 2003] (DOI: 10.1109/TPAMI.2003.1177156)
 """
-function feature_transform(I::AbstractArray{Bool,N}, w::Union{Nothing,NTuple{N}}=nothing) where N
+function feature_transform(img::AbstractArray{<:Union{Bool,AbstractGray{Bool}},N};
+                           weights::Union{Nothing,NTuple{N}}=nothing,
+                           nthreads::Int = length(img) < 1000 ? 1 : Threads.nthreads()) where N
+    nthreads > 0 || error("the number of threads must be positive")
+    N == 0 && return reshape([CartesianIndex()])
+    # Allocate the output
+    F = similar(img, CartesianIndex{N})
+    axsimg = axes(img)
     # To allocate temporary storage for voronoift!, compute one
     # element (so we have the proper type)
-    fi = first(CartesianIndices(axes(I)))
-    drft = DistRFT(fi, w, (), Base.tail(fi.I))
-    tmp = Vector{typeof(drft)}()
-
-    # Allocate the output
-    F = similar(I, CartesianIndex{N})
-
-    # Compute the feature transform (recursive algorithm)
-    computeft!(F, I, CartesianIndex(()), w, tmp)
-
-    F
+    fi = first(CartesianIndices(axsimg))
+    drft = DistRFT(fi, weights, (), Base.tail(fi.I))
+    if nthreads == 1
+        tmp = typeof(drft)[]
+        computeft!(F, img, axsimg, CartesianIndex(()), weights, tmp)
+        # Finish the last dimension (for threading, we avoid doing it in computeft!)
+        finishft!(F, img, axsimg, CartesianIndex(()), weights, tmp)
+    else
+        tasks = Task[]
+        tmps = [typeof(drft)[] for _ = 1:nthreads]
+        saxs = SplitAxes(axsimg, nthreads - 0.2)   # give main thread less work since it also schedules the others
+        for i = nthreads:-1:2
```

Comment:
What's the difference between `nthreads:-1:2` and `2:nthreads` or `1:nthreads-1` here? My understanding is that we can't expect the for-loop to assign the i-th job to the i-th threads.

Reply:
You're right. I'll change it.